### DISCUSSION TITLE
Question about 'Trial level node is missing the "type" parameter. '

### DISCUSSION THREAD

**Discussion by chguw** [jspsych/jsPsych#3015] [Q&A]
I want to construct an experiment using four blocks:

block_up, block_up, block_down, block_down.

Within each block there are several trials, within each trial, there would be a fixation followed by a standard volume sound and a follow-up sound, then a prompt saying "the second audio volume is _____ then the first audio volume", and waiting for the participants to click the buttom. If the participant click the "High" buttom in the block_up block for example, then the block ends, and move on to the next block.

I have coded the experiment, but when I run it, the console shows 'Trial level node is missing the "type" parameter', I am quite confused, I have no idea what the problem it is, and there is no manual addressing the problem, could you please tell me what's wrong with my code, and how I can debug it, thank you!

`
<!DOCTYPE html>
<html>

    <head>
        <title>My experiment 2</title>
        <script src="jspsych/jspsych.js"></script>
        <script src="jspsych/plugin-audio-keyboard-response.js"></script>
        <script src="jspsych/plugin-html-keyboard-response.js"></script>
        <script src="jspsych/plugin-html-button-response.js"></script>
        <script src="jspsych/plugin-preload.js"></script>
        <link href="jspsych/jspsych.css" rel="stylesheet" type="text/css"/>
    </head>

    <body></body>

    <script>
        var timeline = [];

        var jsPsych = initJsPsych({
            on_finish: function() {
                jsPsych.data.displayData();
                jsPsych.data.get().localSave("csv", "mydata.csv")
            }
        });

        var pure_tones_up = [
            {pures: 'sounds/pure_tone_0.010.wav'},
            {pures: 'sounds/pure_tone_0.012.wav'},
            {pures: 'sounds/pure_tone_0.014.wav'},
            {pures: 'sounds/pure_tone_0.016.wav'},
            {pures: 'sounds/pure_tone_0.018.wav'},
            {pures: 'sounds/pure_tone_0.020.wav'},
            {pures: 'sounds/pure_tone_0.022.wav'},
            {pures: 'sounds/pure_tone_0.024.wav'},
            {pures: 'sounds/pure_tone_0.026.wav'},
            {pures: 'sounds/pure_tone_0.028.wav'},
            {pures: 'sounds/pure_tone_0.030.wav'},
            {pures: 'sounds/pure_tone_0.040.wav'}
        ];

        var pure_tones_down = [
            {pures: 'sounds/pure_tone_0.040.wav'},
            {pures: 'sounds/pure_tone_0.030.wav'},
            {pures: 'sounds/pure_tone_0.028.wav'},
            {pures: 'sounds/pure_tone_0.026.wav'},
            {pures: 'sounds/pure_tone_0.024.wav'},
            {pures: 'sounds/pure_tone_0.022.wav'},
            {pures: 'sounds/pure_tone_0.020.wav'},
            {pures: 'sounds/pure_tone_0.018.wav'},
            {pures: 'sounds/pure_tone_0.016.wav'},
            {pures: 'sounds/pure_tone_0.014.wav'},
            {pures: 'sounds/pure_tone_0.012.wav'},
            {pures: 'sounds/pure_tone_0.010.wav'}
        ];

        var preload = {
            type: jsPsychPreload,
            auto_preload: true,
            audio: ['sounds/NOISE.wav', 'sounds/pure_tone_0.010.wav', 'sounds/pure_tone_0.012.wav', 'sounds/pure_tone_0.014.wav', 'sounds/pure_tone_0.016.wav', 'sounds/pure_tone_0.018.wav', 'sounds/pure_tone_0.020.wav', 'sounds/pure_tone_0.022.wav', 'sounds/pure_tone_0.024.wav', 'sounds/pure_tone_0.026.wav', 'sounds/pure_tone_0.028.wav', 'sounds/pure_tone_0.030.wav', 'sounds/pure_tone_0.040.wav']
        };

        timeline.push(preload);

        var welcome = {
            type: jsPsychHtmlButtonResponse,
            stimulus: 'Welcome to the experiment. <br>Press the buttom to continue.',
            choices: ['Continue']
        };

        timeline.push(welcome);

        var instructions = {
            type: jsPsychHtmlButtonResponse,
            stimulus: `
            <p>instructions</p>
            <p>Pess any key to begin.</p>
            `,
            choices: ['Continue'],
            post_trial_gap: 2000
        };

        timeline.push(instructions);

        var fixation = {
            type: jsPsychHtmlKeyboardResponse,
            stimulus: '<div style = "font-size: 60px;">+</div>',
            choices: 'NO_KEYS',
            trial_duration: 1000
        };

        var sounds_standard = {
            type: jsPsychAudioKeyboardResponse,
            stimulus: 'sounds/pure_tone_0.020.wav',
            choices: 'NO_KEYS',
            trial_duration: 2000
        };

        var sounds_follow = {
            type: jsPsychAudioKeyboardResponse,
            stimulus: jsPsych.timelineVariable('pures'),
            choices: 'NO_KEYS',
            trial_duration: 2000
        };

        var responsewait = {
            type: jsPsychHtmlButtonResponse,
            stimulus: '',
            choices: ['High', 'Equal', 'Low'],
            prompt: 'The volumn of the second audio is ____ then that of first audio',
            post_trial_gap: 500
        }

        var trial_up = {
            timeline: [fixation, sounds_standard, sounds_follow],
            timeline_variables: pure_tones_up,
            data: {
                Trial: 'Trial_Up'
            },
            repetitions: 12,
            on_finish: function(data) {
                if (data.response < 2) {
                    jsPsych.endCurrentTimeline();
                }
            }
        }

        var trial_down = {
            timeline: [fixation, sounds_standard, sounds_follow],
            timeline_variables: pure_tones_down,
            data: {
                Trial: 'Trial_Down'
            },
            repetitions: 12,
            on_finish: function(data) {
                if (data.response > 0) {
                    jsPsych.endCurrentTimeline();
                }
            }
        }

        var block_up = {
            timeline: [trial_up],
            data: {
                Block: 'Block_Up'
            }
        }

        var block_down = {
            timeline: [trial_down],
            data: {
                Block: 'Block_Down'
            }
        }

        timeline.push([block_up, block_up, block_down, block_down]);

        jsPsych.run(timeline);
    </script>

</html>
`

**Comment by ChristopheBossens** [comment]
I think in the final timeline push, you have to expand the array so that it actually adds jsPsych trials and not the actual array (which causes the error because the array does not have parameter defined)

`timeline.push(...[block_up, block_up, block_down, block_down]);`
